Spatial_Engine/Create/ShapeProfiles/TaperFlangeISectionProfile.cs

```diff
@@ -0,0 +1,188 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Collections.ObjectModel;
+using BH.oM.Spatial.ShapeProfiles;
+using BH.oM.Geometry;
+using BH.oM.Geometry.CoordinateSystem;
+using System;
+using BH.Engine.Reflection;
+using BH.oM.Reflection.Attributes;
+using BH.Engine.Geometry;
+using System.ComponentModel;
+
+namespace BH.Engine.Spatial
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a I-shaped profile based on input dimensions. Method generates edge curves based on the inputs.")]
+        [InputFromProperty("height")]
+        [InputFromProperty("width")]
+        [InputFromProperty("webThickness")]
+        [InputFromProperty("flangeThickness")]
+        [InputFromProperty("flangeSlope")]
+        [InputFromProperty("rootRadius")]
+        [InputFromProperty("toeRadius")]
+        [Output("I", "The created ISectionProfile.")]
+        public static TaperFlangeISectionProfile TaperFlangeISectionProfile(double height, double width, double webThickness, double flangeThickness, double flangeSlope, double rootRadius, double toeRadius)
+        {
+            if (height < flangeThickness * 2 + rootRadius * 2 || height <= flangeThickness * 2)
+            {
+                InvalidRatioError("height", "flangeThickness and rootRadius");
+                return null;
+            }
+
+            if (width < webThickness + rootRadius * 2 + toeRadius * 2)
+            {
+                InvalidRatioError("width", "webthickness, rootRadius and toeRadius");
+                return null;
+            }
+
+            if (flangeSlope < 0)
+            {
+                Reflection.Compute.RecordError("Flange slope must be positive. Values typically range from 0 to 1/6");
+                return null;
+            }            
+
+            if (flangeSlope > Math.Atan(4 * flangeThickness / width))
+            {
+                InvalidRatioError("Width", "FlangeThickness");
+                return null;
+            }
+
+            if (toeRadius > flangeThickness - width / 4 * Math.Tan(flangeSlope))
```

Comment:
Think it should be like this now?
```suggestion
            if (toeRadius > flangeThickness - width / 4 * flangeSlope)
```